src/pages/AddTask.tsx

```diff
@@ -102,7 +103,27 @@ const AddTask = (props: AddTaskProps) => {
             color: colors.filter((color) => color.selected)[0].color,
         }
 
-        addTask(task).catch((err) => { alert(JSON.stringify(err)) }).then(() => props.goBack());
+        let taskId: Database['public']['Tables']['Tasks']['Row']['task_id'];
+
+        try {
+            taskId = await addTask(task);
+            props.goBack();
+        } catch (err) {
+            console.log(err);
+            props.goBack();
+            return;
+        }
+
+        await Promise.all(selectedTags.map(async (tag) => {
+            const taskWithTag: Database['public']['Tables']['TasksWithTags']['Insert'] = {
+                task_id: taskId,
+                tag_name: tag.tag_name,
+            }
+            return await addTaskWithTags(taskWithTag);
+        })).catch((err) => {
+            console.log(err);
```

Comment:
also alert user, maybe something like "task created, but could not add all tags. try editing the task and updating appropriately" idk